File: app/src/main/java/activitytest/example/com/roomdemo/home/viewmodel/BlogViewModel.java
package activitytest.example.com.roomdemo.home.viewmodel;

import android.util.Log;

import java.util.List;
import java.util.Set;
import activitytest.example.com.roomdemo.home.HomeRepository;
import activitytest.example.com.roomdemo.home.bean.Root;
import activitytest.example.com.roomdemo.home.lifecycle.HomeLifeCycle;
import activitytest.example.com.roomdemo.main.MainRepository;
import activitytest.example.com.roomdemo.main.activity.MainActivity;
import activitytest.example.com.roomdemo.main.database.Configuration;
import activitytest.example.com.roomdemo.main.database.MyDatabase;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class BlogViewModel extends ViewModel {
    private static final String TAG  = HomeLifeCycle.class.getName ();

    private MutableLiveData<List<Root>> mutableLiveData;
    private HomeRepository homeRepository;

    public void init(){
        if (mutableLiveData != null){
            return;
        }
        homeRepository = HomeRepository.getInstance ();
        mutableLiveData = homeRepository.listBlogInfo ();
    }

    /**
     * 返回博客信息
     * @return 博客信息
     */
    public LiveData<List<Root>> getBlogInfoLiveData() {
        return mutableLiveData;
    }


    /**
     * 返回博文的链接
     * @param blogContent 博文的链接
     * @return 博文内容
     */
    public MutableLiveData<Set<String>> setBlogContent(List<String> blogContent){
        return homeRepository.downLoadBlogContent ( blogContent );
    }

}

File: app/src/main/java/activitytest/example/com/roomdemo/home/viewmodel/BlogViewModel.kt
package activitytest.example.com.roomdemo.home.viewmodel

import activitytest.example.com.roomdemo.home.HomeRepository
import activitytest.example.com.roomdemo.home.bean.Root
import activitytest.example.com.roomdemo.home.lifecycle.HomeLifeCycle
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class BlogViewModel : ViewModel() {
    private var rootLiveData: MutableLiveData<List<Root?>?>? = null
    private var homeRepository: HomeRepository? = null
    fun init() {
        if (rootLiveData != null) {
            return
        }
        homeRepository = HomeRepository.instance
        this.rootLiveData = homeRepository?.listBlogInfo()
    }

    /**
     * 返回博客信息
     * @return 博客信息
     */
    val blogInfoLiveData: LiveData<List<Root?>?>?
        get() = rootLiveData

    /**
     * 返回博文的链接
     * @param blogContent 博文的链接
     * @return 博文内容
     */
    fun listBlogContent(blogContent: List<String?>): MutableLiveData<Set<String>> {
        return homeRepository!!.downLoadBlogContent(blogContent)
    }

    companion object {
        private val TAG = HomeLifeCycle::class.java.name
    }
}